### DISCUSSION TITLE
model.update causing Next application to crash

### DISCUSSION THREAD

**Discussion by dextermb** [prisma/prisma#10720] [Q&A] [answered]
Hey all

I've just started tinkering with PlanetScale and Prisma for my Next application. Unfortunately I've come across an issue where calling `<model>.update` seems to crash my Next application (at least running it locally).

Here's a snippet of the crash I'm getting:

```
wait  - compiling /api/welcome...
event - compiled client and server successfully in 181 ms (1039 modules)
error Command failed with signal "SIGBUS".
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
```

It doesn't really give me much information other than `error Command failed with signal "SIGBUS".`. I've had a look through my `/api/welcome` handler by removing sections of code until it fails and unfortunately it seems to happen when I attempt to update a model.

```js
await database.user.update({
  where: { id },
  data: {
    name: data.name,
    org: {
      create: {
        name: data.org
      }
    }
  }
})
```

At first I thought it might be the nested relation but that doesn't seem to be the issue, I also have a different endpoint (`/api/auth/me`) that uses `<model>.findUnique` perfectly fine.

Here's the Gist for the whole file, although to me it doesn't seem like I'm trying to do anything weird:
https://gist.github.com/dextermb/a1c797ffa85e254bcbe67b8c0153999a

Any thoughts on what might be going on?

**Comment by dextermb** [comment]
To further this being an issue with `<model>.update` I replaced the `update` call with another method, such as `findUnique`, and as expected the endpoint did not crash.

I've also double checked `data` and it only contains simple strings:
```
{
  name: 'Dexter',
  org: 'Acme'
}
```

Very weird!

**Reply by dextermb** [reply]
I noticed we can enable debugging for Prisma, here's the output of that:
```
prisma:client:libraryEngine sending request, this.libraryStarted: true +71ms
  prisma:client Prisma Client call: +35ms
  prisma:client prisma.user.update({
  prisma:client   where: {
  prisma:client     id: 'ckx83iboo0008vjpgmjls82d1'
  prisma:client   },
  prisma:client   data: {
  prisma:client     name: 'Dexter',
  prisma:client     org: {
  prisma:client       create: {
  prisma:client         name: 'Acme'
  prisma:client       }
  prisma:client     }
  prisma:client   }
  prisma:client }) +1ms
  prisma:client Generated request: +0ms
  prisma:client mutation {
  prisma:client   updateOneUser(
  prisma:client     where: {
  prisma:client       id: "ckx83iboo0008vjpgmjls82d1"
  prisma:client     }
  prisma:client     data: {
  prisma:client       name: "Dexter"
  prisma:client       org: {
  prisma:client         create: {
  prisma:client           name: "Acme"
  prisma:client         }
  prisma:client       }
  prisma:client     }
  prisma:client   ) {
  prisma:client     id
  prisma:client     name
  prisma:client     email
  prisma:client     emailVerified
  prisma:client     image
  prisma:client     orgId
  prisma:client     teamId
  prisma:client     leadId
  prisma:client   }
  prisma:client }
  prisma:client  +0ms
  prisma:client:libraryEngine sending request, this.libraryStarted: true +35ms
error Command failed with signal "SIGBUS".
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
```

**Reply by dextermb** [reply]
Interestingly it also crashes on `<model>.updateMany`:

```
prisma:client:libraryEngine sending request, this.libraryStarted: true +61ms
  prisma:client Prisma Client call: +31ms
  prisma:client prisma.user.updateMany({
  prisma:client   where: {
  prisma:client     id: 'ckx83iboo0008vjpgmjls82d1'
  prisma:client   },
  prisma:client   data: {
  prisma:client     name: 'Dexter'
  prisma:client   }
  prisma:client }) +0ms
  prisma:client Generated request: +0ms
  prisma:client mutation {
  prisma:client   updateManyUser(
  prisma:client     where: {
  prisma:client       id: "ckx83iboo0008vjpgmjls82d1"
  prisma:client     }
  prisma:client     data: {
  prisma:client       name: "Dexter"
  prisma:client     }
  prisma:client   ) {
  prisma:client     count
  prisma:client   }
  prisma:client }
  prisma:client  +0ms
  prisma:client:libraryEngine sending request, this.libraryStarted: true +31ms
error Command failed with signal "SIGBUS".
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
```

**Comment by dextermb** [comment]
I've made some progress in terms of finding the error, but this time it doesn't seem to crash the endpoint but instead just error out.

Here you can see `user.findUnique` using `id` within the `where` clause that works absolutely fine:
```
  prisma:client:libraryEngine sending request, this.libraryStarted: true +783ms
  prisma:client Prisma Client call: +61ms
  prisma:client prisma.user.findUnique({
  prisma:client   where: {
  prisma:client     id: 'ckx83iboo0008vjpgmjls82d1'
  prisma:client   }
  prisma:client }) +1ms
  prisma:client Generated request: +0ms
  prisma:client query {
  prisma:client   findUniqueUser(where: {
  prisma:client     id: "ckx83iboo0008vjpgmjls82d1"
  prisma:client   }) {
  prisma:client     id
  prisma:client     name
  prisma:client     email
  prisma:client     emailVerified
  prisma:client     image
  prisma:client     orgId
  prisma:client     teamId
  prisma:client     leadId
  prisma:client   }
  prisma:client }
  prisma:client  +0ms
  prisma:client:libraryEngine sending request, this.libraryStarted: true +62ms
```

But when `id` is used in `user.update` it doesn't work and instead errors:
```
  prisma:client Prisma Client call: +29ms
  prisma:client prisma.user.update({
  prisma:client   where: {
  prisma:client     id: 'ckx83iboo0008vjpgmjls82d1'
  prisma:client   },
  prisma:client   data: {
  prisma:client     name: 'Dexter'
  prisma:client   }
  prisma:client }) +0ms
  prisma:client Generated request: +0ms
  prisma:client mutation {
  prisma:client   updateOneUser(
  prisma:client     where: {
  prisma:client       id: "ckx83iboo0008vjpgmjls82d1"
  prisma:client     }
  prisma:client     data: {
  prisma:client       name: "Dexter"
  prisma:client     }
  prisma:client   ) {
  prisma:client     id
  prisma:client     name
  prisma:client     email
  prisma:client     emailVerified
  prisma:client     image
  prisma:client     orgId
  prisma:client     teamId
  prisma:client     leadId
  prisma:client   }
  prisma:client }
  prisma:client  +0ms
  prisma:client:libraryEngine sending request, this.libraryStarted: true +29ms
  prisma:client:fetcher Error: Error occurred during query execution:
  prisma:client:fetcher ConnectorError(ConnectorError { user_facing_error: None, kind: QueryError(Server(ServerError { code: 1105, message: "symbol db.`User`.id not found", state: "HY000" })) })
  prisma:client:fetcher     at prismaGraphQLToJSError (/Users/dexter/projects/next/tw-3/node_modules/@prisma/client/runtime/index.js:35958:10)
  prisma:client:fetcher     at Object.request (/Users/dexter/projects/next/tw-3/node_modules/@prisma/client/runtime/index.js:36319:17)
  prisma:client:fetcher     at async cb (/Users/dexter/projects/next/tw-3/node_modules/@prisma/client/runtime/index.js:38653:26)
  prisma:client:fetcher     at async handler (webpack-internal:///./pages/api/welcome.js:26:9)
  prisma:client:fetcher     at async Object.apiResolver (/Users/dexter/projects/next/tw-3/node_modules/next/dist/server/api-utils.js:102:9)
  prisma:client:fetcher     at async DevServer.handleApiRequest (/Users/dexter/projects/next/tw-3/node_modules/next/dist/server/next-server.js:1064:9)
  prisma:client:fetcher     at async Object.fn (/Users/dexter/projects/next/tw-3/node_modules/next/dist/server/next-server.js:951:37)
  prisma:client:fetcher     at async Router.execute (/Users/dexter/projects/next/tw-3/node_modules/next/dist/server/router.js:222:32)
  prisma:client:fetcher     at async DevServer.run (/Users/dexter/projects/next/tw-3/node_modules/next/dist/server/next-server.js:1135:29)
  prisma:client:fetcher     at async DevServer.run (/Users/dexter/projects/next/tw-3/node_modules/next/dist/server/dev/next-dev-server.js:445:20) +1m
error - Error: 
Invalid `prisma.user.update()` invocation:


  Error occurred during query execution:
ConnectorError(ConnectorError { user_facing_error: None, kind: QueryError(Server(ServerError { code: 1105, message: "symbol db.`User`.id not found", state: "HY000" })) })
```

`User.id` is both in the schema, type hinting, and database.

Unsure why the database is saying that it doesn't exist. Perhaps I need to take this up with PlanetScale.
<img width="621" alt="image" src="https://user-images.githubusercontent.com/6137789/146357616-b93c7be1-b0a0-479f-9dda-4b8204d67c29.png">

**Reply by janpio** [reply]
This is indeed much more the expected behavior instead of the weird SIGBUS crash.

If you find a way to reproduce the crash, please let us know. That one would definitely be something we need to look into. The database returning a weird and unexpected error on the other hand is more something for PlanetScale support, as you concluded yourself.

**Reply by dextermb** [reply]
Hey @janpio after chatting to PlanetScale support it actually seems like Prisma was trying to execute a weird query, you can find the Gist here: https://gist.github.com/dextermb/2698971c153ba74b339879363295123b.

Here's an extract:
```
Query: SELECT `db`.`User`.`id`, `db`.`OrgTeamAdmin`.`teamId`, `db`.`OrgTeamAdmin`.`userId` FROM `db`.`OrgTeamAdmin` WHERE 1=0
Duration: 29ms
Query: ROLLBACK
Duration: 30ms
```

I've been able to re-produce why this is happening in my application. Here's a snippet of my schema:

```prisma
model User {
  id String @id @default(cuid())
  name String?
  email String? @unique
  emailVerified DateTime?
  image String?

  accounts Account[]
  sessions Session[]

  orgId String?
  org Org? @relation(fields: [orgId], references: [id])

  teamId String?
  team OrgTeam? @relation(fields: [teamId], references: [id])

  leadId String?
}

model OrgTeamAdmin {
  teamId String
  team OrgTeam @relation(fields: [teamId], references: [id])

  userId String
  // user User @relation(fields: [userId], references: [id])

  @@unique([teamId, userId])
}
```

The `user` relation (currently commented out) on `OrgTeamAdmin` seems to make the weird select. Not sure why.

**Reply by janpio** [reply]
Spun this out into its own issue at https://github.com/prisma/prisma/issues/10822

**Comment by dextermb** [comment, marked as answer]
~I've opened a request with PlanetScale support. Hopefully they'll know what's going on.~

Edit: See #10822

**Comment by nechmads** [comment]
Hi. Everything worked fine for us until yesterday and now I see the exact same thing on one of our models.
It's really weird as if I change the update call to update same field on another model everything works fine. It just crashed with this specific model.
I tried to revert to Prisma 3.5 but it seems that what ever I do, the prisma client version that is used is 3.7. Tried to delete node_modules, remove from global, add to global... I always get the same error and it comes from prisma client version 3.7
I also tried to reset the db, remove all tables and run migration again. Nothing helps. Same error with the same model and just with that model. And as stated above same error even if I do UpdateMany with no where clause which is eve weirder.

**Reply by janpio** [reply]
Did you upgrade yesterday and were running 3.5 before?

> I tried to revert to Prisma 3.5 but it seems that what ever I do, the prisma client version that is used is 3.7. Tried to delete node_modules, remove from global, add to global... I always get the same error and it comes from prisma client version 3.7

This should not really be possible and points to some weird caching or other complication happening in your project.

**Comment by nechmads** [comment]
Found the solution but not sure why it was needed.
One of the relations pointing to the model with the problem didn't have "onDelete: NoAction, onUpdate: NoAction" on it.
But this is weird, as we use MySql and based on the docs there is no automatic updates on relations.
Anyways - thanks @dextermb for your post here. Definitely helped me find the solution.

**Reply by janpio** [reply]
Are you using MySQL or PlanetScale? That makes quite a difference in the referential actions handling.

**Reply by nechmads** [reply]
Yes. On PlanetScale. Is there any docs on what are the differences we should know about?

**Reply by janpio** [reply]
https://github.com/prisma/prisma/issues/7292 and all the discussion beneath - as things are currently figured out, this did not make it into our full docs yet. PlanetScale itself might also have some docs.

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
